Makefile

```diff
@@ -41,11 +41,48 @@ license-check: $(VENV) deps
 .PHONY: lint
 lint: $(VENV) deps
 	@echo "LINTING"
-	$(PYTHON) -m flake8 tests detection_rules --ignore D203 --max-line-length 120
+	$(PYTHON) -m flake8 tests detection_rules --ignore D203,N815 --max-line-length 120
 
 .PHONY: test
 test: $(VENV) lint pytest
 
+.PHONY: test-cli
+test-cli: $(VENV)
+	@echo "Running detection-rules CLI tests..."
+	@echo "Refreshing redirect mappings in ATT&CK"
+	@$(PYTHON) -m detection_rules dev attack refresh-redirect-mappings
+	@echo "Viewing rule: threat_intel_indicator_match_address.toml"
+	@$(PYTHON) -m detection_rules view-rule rules/cross-platform/threat_intel_indicator_match_address.toml
+	@echo "Exporting rule by ID: 0a97b20f-4144-49ea-be32-b540ecc445de"
+	@$(PYTHON) -m detection_rules export-rules --rule-id 0a97b20f-4144-49ea-be32-b540ecc445de
+	@echo "Updating rule data schemas"
+	@$(PYTHON) -m detection_rules dev schemas update-rule-data
+	@echo "Validating rule: execution_github_new_event_action_for_pat.toml"
+	@$(PYTHON) -m detection_rules validate-rule rules_building_block/execution_github_new_event_action_for_pat.toml
+	@echo "Checking licenses"
+	@$(PYTHON) -m detection_rules dev license-check
+	@echo "Building release and updating version lock"
+	@$(PYTHON) -m detection_rules dev build-release --update-version-lock
+	@echo "Refreshing ATT&CK data"
+	@$(PYTHON) -m detection_rules dev attack refresh-data
+	@echo "Updating rules with latest ATT&CK data"
+	@$(PYTHON) -m detection_rules dev attack update-rules
+	@echo "Getting target branches"
+	@$(PYTHON) -m detection_rules dev utils get-branches
+	@echo "Showing latest compatible version for security_detection_engine with stack version 8.12.0"
+	@$(PYTHON) -m detection_rules dev integrations show-latest-compatible --package endpoint --stack_version 8.12.0
+	@echo "Building limited rules for stack version 8.12"
+	@$(PYTHON) -m detection_rules build-limited-rules --stack-version "8.12" --output-file "output_file.ndjson"
+	@echo "Building limited rules for stack version 8.12 with custom rules"
+	@$(PYTHON) -m detection_rules generate-rules-index --overwrite
+	@echo "Building manifests for integrations"
+	@$(PYTHON) -m detection_rules dev integrations build-manifests -i endpoint
+	@echo "Building schemas for integrations"
+	@$(PYTHON) -m detection_rules dev integrations build-schemas -i endpoint
+	@echo "Detection-rules CLI tests completed!"
```

Comment:
we should also add an additional command that does remote commands on a stack (which would be predicated on user config or envvar). We wouldn't run them in CI, just manual. Main reason are the elasticsearch and related dependencies

Reply:
Yea that makes sense. I'll add one.